base/string_utils.hpp

```diff
@@ -306,6 +306,11 @@ void Tokenize(string const & str, char const * delims, TFunctor && f)
   }
 }
 
+/// Splits a string by the delimiter, keeps empty parts, on an empty string returns an empty vector.
+/// Does not support quoted columns, newlines in columns and escaped quotes.
+/// @return false if the line is empty or number of columns differs from |columns|.
+bool ParseCSVRow(string const & s, vector<string> & target, char const delimiter = ',', size_t const columns = 0);
```

Comment:
The parameter has a default value, so the API is generic enough, with only two required parameters for input and output. The `columns` parameter simplifies checks, so a caller needs only to check the return value of the function.